settings.js

```diff
@@ -109,6 +110,18 @@ function _getAppsInsideFolder(installedApps, folder) {
     return appIds;
 }
 
+function _addIconItem(pages, itemId, index, itemsPerPage) {
+    const page = Math.floor(index / itemsPerPage);
+
+    if (page === pages.length)
+        pages.push({});
+
+    pages[page][itemId] = {
+        position: GLib.Variant.new_int32(index % itemsPerPage),
+    };
+    return index + 1;
```

Comment:
Minor nitpick: instead of returning the new index, call it as `_addIconItem(..., index++, ...)`